### PR TITLE
fixed issue for lineage description

### PR DESCRIPTION
**Pull request by 07Himank**
I worked on ... because ... lineage edge description issue fixed, and added support for pipeline as edge for topic entity
fix: #8560 , #8533
### Type of change:

- [x ] Bug fix

### CODE REVIEW

**Review by mohityadav766 — DISMISSED** [review]

**Review by ulixius9 — DISMISSED** [review]

**Comment by 07Himank** [comment]
> @07Himank can you add tests around this.

Yes @harshach I will add it.

File: openmetadata-spec/src/main/resources/json/schema/type/entityLineage.json [hunk 1]
```diff
@@ -124,6 +128,10 @@
         "$ref": "#/definitions/edge"
       },
       "default": null
+    },
+    "description" :{
+      "description": "description of lineage",
+      "type": "string"
```
Comment (ulixius9) [inline openmetadata-spec/src/main/resources/json/schema/type/entityLineage.json:134]: This description field I believe is not required as we are already getting the description in the lineageDetails object, so this looks like an ambiguous field

Comment (07Himank) [reply openmetadata-spec/src/main/resources/json/schema/type/entityLineage.json]: done

File: openmetadata-service/src/test/java/org/openmetadata/service/resources/lineage/LineageResourceTest.java [hunk 2]
```diff
@@ -327,6 +327,16 @@ void putLineageFromDashboardDataModelToTable() throws HttpResponseException {
         "Column level lineage is only allowed between two tables or from table to dashboard.");
   }
 
+  @Order(5)
+  @Test
+  void put_lineageWithDescription() throws HttpResponseException {
```
Comment (harshach) [inline openmetadata-service/src/test/java/org/openmetadata/service/resources/lineage/LineageResourceTest.java:332]: @07Himank please add a test for patch as well

Comment (07Himank) [reply openmetadata-service/src/test/java/org/openmetadata/service/resources/lineage/LineageResourceTest.java:332]: Yeah sure will do it.

File: openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/LineageRepository.java [hunk 3]
```diff
@@ -90,14 +90,13 @@ private String validateLineageDetails(EntityReference from, EntityReference to,
...
-          "Column level lineage is only allowed between two tables or from table to dashboard.");
-    }
-
-    Table fromTable = dao.tableDAO().findEntityById(from.getId());
-    ColumnsEntityInterface toTable = getToEntity(to);
-    if (columnsLineage != null) {
+    if (columnsLineage != null && !columnsLineage.isEmpty()) {
+      if (areValidEntities(from, to)) {
+        throw new IllegalArgumentException(
+            "Column level lineage is only allowed between two tables or from table to dashboard.");
```
Comment (harshach) [inline openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/LineageRepository.java:96]: we don't need to check this. You can have lineage between table to chart as well, topic to table, pipeline to table etc..

Comment (07Himank) [reply openmetadata-service/src/main/java/org/openmetadata/service/jdbi3/LineageRepository.java:96]: @harshach this is for the column level lineage.

**Review by mohityadav766 — APPROVED** [review]

**Pull request merged**